src/main/java/org/fundacion/jala/converter/view/MainInterface.java

```diff
@@ -47,29 +71,138 @@
+        mainButtonAction(videoConverterBtn);
+        mainButtonAction(audioConverterBtn);
+        mainButtonAction(textExtractorBtn);
+        mainButtonAction(metaDataBtn);
+    }
+
+    /**
+     * Sets Panels and positions for elements in Main frame.
+     */
+    public void setPanels() {
+        containerPanel = new JPanel();
+        containerPanel.setLayout(new BorderLayout());
+        containerPanel.setBorder(MARGIN_SPACE);
+        containerPanel.add(TITLE_APP, BorderLayout.NORTH);
+        containerPanel.add(mainPanel, BorderLayout.CENTER);
+        containerPanel.add(COPYRIGHT, BorderLayout.SOUTH);
+        buttonsPanel = new JPanel();
+        buttonsPanel.setLayout(new GridBagLayout());
+        buttonsPanel.setBackground(DARK_BACKGROUND_BTN);
+        setPosition(buttonsPanel, compilerBtn, false);
+        setPosition(buttonsPanel, videoConverterBtn, false);
+        setPosition(buttonsPanel, audioConverterBtn, false);
+        setPosition(buttonsPanel, textExtractorBtn, false);
+        setPosition(buttonsPanel, metaDataBtn, true);
+    }
+
+    /**
+     * Sets positions for FlowLayout.
+     */
+    public void setPosition(JPanel mainPanel, BtnStyle mainBtn, boolean lastBtn) {
+        GridBagConstraints positionConstraint = new GridBagConstraints();
+        positionConstraint.gridx = positionXMainBtn;
+        positionConstraint.gridy = positionYMainBtn;
+        positionConstraint.fill = GridBagConstraints.HORIZONTAL;
+        mainPanel.add(mainBtn, positionConstraint);
+        if (lastBtn) {
+            positionConstraint.gridy = 10;
+            positionConstraint.gridwidth = GridBagConstraints.REMAINDER;
+            positionConstraint.weighty = 1;
+            mainPanel.add(new JLabel(), positionConstraint);
+        }
+        positionYMainBtn += 1;
+    }
+
+    /**
+     * Sets the correct panel according to selected Button.
+     */
+    public void switchPanels(JPanel panel) {
+        mainPanel.removeAll();
+        mainPanel.add(panel);
+        mainPanel.repaint();
+        mainPanel.revalidate();
+    }
+
+    /**
+     * Call the panel for the Button option.
+     *
+     * @param e Action of selected JButton.
+     */
+    @Override
+    public void actionPerformed(ActionEvent e) {
+        if (e.getSource() == compilerBtn) {
+            switchPanels(compilerPanel);
+        }
+        if (e.getSource() == videoConverterBtn) {
+            switchPanels(videoPanel);
+        }
+        if (e.getSource() == audioConverterBtn) {
+            switchPanels(audioPanel);
+        }
+        if (e.getSource() == textExtractorBtn) {
+            switchPanels(textExtractorPanel);
+        }
+        if (e.getSource() == metaDataBtn) {
+            switchPanels(metaDataPanel);
+        }
+    }
+
+    /**
+     * Adds the style hover and out for JButton.
+     *
+     * @param button to add style.
+     */
+    public BtnStyle mainButtonAction(BtnStyle button) {
+        MouseListener ml = new MouseAdapter() {
```

Comment:
The variable should have a more descriptive name.

Reply:
Solved!